services/src/main/java/org/keycloak/protocol/oidc/installation/KeycloakOIDCClientInstallation.java

```diff
@@ -87,7 +87,7 @@ public static boolean showClientCredentialsAdapterConfig(ClientModel client) {
             return false;
         }
 
-        if (client.isBearerOnly() && client.getNodeReRegistrationTimeout() <= 0) {
+        if (client.isBearerOnly() && !client.isServiceAccountsEnabled() && client.getNodeReRegistrationTimeout() <= 0) {
```

Comment:
Seems OK, but could you add tests to https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/client/InstallationTest.java please?